### PR TITLE
Phase2 - HGCal - Filtering PFRecHits based on SNR threshold

### PR DESCRIPTION
**Pull request by felicepantaleo**
This PR is another step towards making RealisticSimClusters more realistic. 

A HGCRecHit produces a PFRecHit if its energy is above 5 sigma noise. 
The quality test is disabled and can be enabled for testing purposes by uncommenting the code in py configuration.
It will be enabled when RealisticSimClusters will be activated.

@malgeri @rovere @cseez @kpedro88 @vandreev11

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by felicepantaleo** [comment]
Is the error connected to the PR?

**Comment by perrotta** [comment]
please test

@felicepantaleo : validation stuffs in 2017 wfs are unlikely related to this PR...
Let retry launching the tests, hopefully in a cleaner IB

**Comment by kpedro88** [comment]
+1

File: RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitWorkerSimple.cc [hunk 1]
```diff
@@ -99,7 +99,7 @@ bool HGCalRecHitWorkerSimple::run(const edm::Event & evt, const HGCUncalibratedR
         hid = detid;
         thickness = ddds_[hid.subdetId() - 3]->waferTypeL(hid.wafer());
         sigmaNoiseGeV = 1e-3 * weights_[layer] * rcorr_[thickness]
-                    * hgcHEF_noise_fC_[thickness - 1] / hgcHEF_fCPerMIP_[thickness - 1];
+                    * hgcEE_noise_fC_[thickness - 1] / hgcEE_fCPerMIP_[thickness - 1];
```
Comment (perrotta) [inline RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitWorkerSimple.cc]: This is just a bug fix for a typo, I would say...

By the way, I think you should have a "break" also after next L111, unless you want to treat "case HcalEndcap:" and "case HGCHEB:" the same: please check

Comment (felicepantaleo) [inline RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitWorkerSimple.cc]: @perrotta yes, this is a bugfix for a typo and I confirm that the two cases should be treated the same.

**Comment by perrotta** [comment]
+1
A new class PFRecHitQTestHGCalThresholdSNR is created which allows cutting on signalOverSigmaNoise
Default value is 0: no effect on standard sequences by now

File: RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitHGC_cfi.py [hunk 2]
```diff
@@ -21,21 +21,33 @@
...
+#                     thresholdSNR = cms.double(5.0),
+#                    ),                 
                 )
            ),
            cms.PSet(
              name = cms.string("PFHGCHEBRecHitCreator"),
              src  = cms.InputTag("HGCalRecHit:HGCHEBRecHits"),
              geometryInstance = cms.string(""),
              qualityTests = cms.VPSet(
+#                 cms.PSet(
```
Comment (slava77) [inline RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitHGC_cfi.py]: shouldn't these have regular comments to clarify why the code is commented out?

Comment (perrotta) [inline RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitHGC_cfi.py]: Indeed...  Even if it is just supposed to stay as such only temporarily, better to have a line explaining what the commented out config is meant to, for documentation purposes.
Removed the "+1" while waiting for those comments

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
A new class PFRecHitQTestHGCalThresholdSNR is created which allows cutting on signalOverSigmaNoise
Default value is 0: no effect on standard sequences by now

**Comment by kpedro88** [comment]
+1

**Comment by davidlange6** [comment]
merge

**Pull request merged**